ra/ra.go

```diff
@@ -1754,6 +1755,51 @@ func (ra *RegistrationAuthorityImpl) UpdateRegistration(ctx context.Context, req
 	return update, nil
 }
 
+// UpdateRegistrationContact updates an existing Registration's contact.
+func (ra *RegistrationAuthorityImpl) UpdateRegistrationContact(ctx context.Context, req *rapb.UpdateRegistrationContactRequest) (*corepb.Registration, error) {
+	if req == nil || req.RegistrationID == 0 {
+		return nil, errIncompleteGRPCRequest
+	}
+
+	err := ra.validateContacts(req.Contact)
+	if err != nil {
+		return nil, err
+	}
+
+	update, err := ra.SA.UpdateRegistrationContact(ctx, &sapb.UpdateRegistrationContactRequest{
+		RegistrationID: req.RegistrationID,
+		Contact:        req.Contact,
+	})
+	if err != nil {
+		// berrors.InternalServerError since the user-data was validated before being
+		// passed to the SA.
+		err = berrors.InternalServerError("Could not update registration: %s", err)
```

Comment:
Suggesting two changes here (and the same in UpdateRegistrationKey):
- lowercase the error string, since that's best practice
- use fmt.Errorf because it does a better job of preserving the inner error (using %w instead of %s) for inspection in tests and other places, and we know that any fmt.Errorf will get converted into InternalServerError at the WFE anyway.
```suggestion
		err = fmt.Errorf("failed to update registration contact: %w", err)
```

Reply:
What do you think about using `fmt.Errorf` in the SA as well? I'm assuming maybe not, because the logic in the SA doesn't (currently) preserve any inner error and does make use of a specific error type in one case.

Reply:
Yep, I concur.